torch/_higher_order_ops/flex_attention.py

```diff
@@ -416,7 +416,7 @@ def flex_attention_functionalize(
     assert isinstance(score_mod_other_buffers_unwrapped, tuple)
     assert isinstance(mask_mod_other_buffers_unwrapped, tuple)
     assert all(
-        isinstance(item, torch.Tensor)
+        isinstance(item, (torch.Tensor, torch.SymInt))
```

Comment:
Do you have a handle on when you need to test for int vs when you need to test for SymInt and int? It's very muddy to me and if you actually can enforce some stronger invariant here I would like to know about it.

Reply:
+1 on invariants. @ydwu4 is it possible to make it so that we are never dealing with `int`? Either via promoting the ints to SymInts or baking the ints into the traced graph

Reply:
Yeah, for all compiled hops, the invariants are:
1. For Dispatch keys that can be triggered during tracing: e.g. functionalization, fake, proxy, autograd, the other_buffers need t handle tensors and SymInts.
2. For keys that can be triggered during eager run( e.g. torch.compile(backend="eager") or no torch.compile), other_buffers need to handle tensors and ints.

For flex_attention, lifted ints/symints can only appear in score_mod_other_buffers and mask_mod_other_buffers when there're closed over tensors (because the mask and score mod are traced with scalar tensors).

I can add a note to state this invariant clearly.